endpoint.go

```diff
@@ -637,6 +639,14 @@ func EndpointOptionGeneric(generic map[string]interface{}) EndpointOption {
 	}
 }
 
+// CreateOptionIpam function returns an option setter for the ipam configuration for this endpoint
+func CreateOptionIpam(prefAddress net.IP, ipamOptions map[string]string) EndpointOption {
```

Comment:
@jc-m agreed that this is a new change and okay to consider multiple ip-addresses. But the use-case (of VIP) is something that is not currently implemented in libnetwork. One of the restrictions that we are thinking of supporting this preferred IP will be that the user-provided ip-address will have to be in one of the subnet and ip-range configured for that network. If the IP falls outside the configured range, it is a failure. Also, once an ip-address is reserved for an endpoint, the same IP cannot be used for other endpoint (as IPAM will reserve that IP). How will it work for VIP ?
These are just adding to @aboch's point of supporting a single IP first and make sure we understand more use-cases better so that we can handle them appropriately.